### DISCUSSION TITLE
DPL speist manchmal bei Solarüberschuß ein. Ein Bug?

### DISCUSSION THREAD

**Discussion by hoschiking** [hoylabs/OpenDTU-OnBattery#713] [Q&A]
Folgendes Setup:
PV-Anlage 3kWp speist per Growatt WR ins Netz ein.
OpenDTU-onBattery  macht den Rest (Speicher laden per MPPT und Einspeisung vom Speicher per HM-300)
Zusätzlich lade ich den Überschuß AC-gestützt in den Speicher (Regeliung per Script im ioBroker)
Soweit funktioniert alles prima.
Manchmal jedoch kommt der DPL aus dem Tritt und speist mit max-Leistung (Powerlimit) ein, obwohl von der großen Anlage genug Überschuß da ist und ich keinen Netzbezug habe und große Leistung >2kW einspeise. Um diese Fehlverhalten zu beenden, muss ich den DPL aus und wieder einschalten. Dies habe ich schon seit mehreren Wochen beobachtet, kann aber nicht wirklich greifen, woran es liegen könnte.  Ich weiß nicht, ob es damit was zu tun hat, es könnte aber zeitlich ganz gut hinkommen, habe ich in meine Regelung folgendes implementiert:
Für die Zeit der AC-Überschußeinspeisung schalte ich den DPL und den HM-300 per MQTT Kommando aus (powerlimiter/cmd/mode auf 1).   Dies ist aufgrund des gesamten Aufbaus der Anlage notwendig und so verhindere ich, dass bei kurzzeitigem Netzbezug der DPL anspringt und wieder einspeist, obwohl ich gerade den Speicher lade. So erschlage ich mehrere Probleme mit einer Klappe.
Gestern zB bei bestem Sonnenschein hatte ich ca 6 Stunden viel Überschuß, ich war nicht zu Hause und dennoch hat sich der DPL für mehrere Stunden eingeschaltet und mit Max-Leistung eingespeist. In der Zeit wurde der DPL weder Aus noch eingeschaltet.
Siehe Screenshot in der Zeit kurz nach 11 Uhr bis kurz nach 13 Uhr, erst per manuelles Eingreifen konnte ich den Spuck beenden.
@schlimmchen 
Eine Idee woran es liegt?
![Screenshot_5](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/159241718/94cebbb7-adeb-4841-896f-751b450b5d4b)

**Comment by schlimmchen** [comment]
Hm, das ist ja Mist.

Hast du den Laderegler per VE.Direct angeschlossen an der OpenDTU-OnBattery? Falls ja kannst du schwören, dass niemand den DPL Mode auf 2 per MQTT gesetzt hat? :monocle_face: 

Das ist aber nur ein Strohhalm...

Wie weit geht das denn da runter dein "P Grid"? War die Einspeisung an irgendeiner magischen Grenze angekommen, als der DPL eingeschaltet hat? Vermutlich nicht, ein Überlauf irgendeines Integers ist hier unwahrscheinlich. Wäre aber eine stimmige Sache. Dürfte aber erst bei gut -16kW Netzbezug überhaupt in Frage kommen.

Was tust du dann genau, wenn du den DPL "beruhigst"? Neustart? Stromlos machen der Platine? MQTT Mode umstellen? DPL aus- und wieder einschalten über die Web UI?

Auch für dich gilt natürlich: Ein Log wäre hier wahrscheinlich sehr wertvoll. Wenn du das nächste man in dieser Situation bist, kannst du bitte eine Minute lang die Console aufmachen in der Web UI der OpenDTU und den Inhalt mit uns teilen? Der DPL spricht ja recht viel, da kann man vielleicht einen Grund erahnen.

**Reply by hoschiking** [reply]
Ja, der Laderegler ist per VE.Direct angeschlossen. Ja, DPL-Mode 2 kann ich defintiv ausschliessen.
P Grid ist in der Zeit (fast) durchgehend bei ca. -2400 bis -2600 W. Interessant ist, das habe ich so eben entdeckt, dass genau um um 11.11 Uhr es einen kurzen Peak ins positive gibt (siehe Screenshot), genau dann springt der DPL an, da in dem Moment mangels Überschuß die AC-Ladung unterbrochen wird, soweit so gut. Leider bleibt er dann an und geht nicht mehr aus, obwohl der sich wieder im Mode 1 befindet. D.h. es scheint so, als ob der beim kurzzeitigen Aus und wieder Einschalten aus dem Tritt kommt. Den kurzen Peak hat mir Grafana bei der Tagesansicht unterschlagen, erst beim starken "reinzommen" wurde der sichtbar. Daher war meine erste Aussage nicht ganz korrekt.
![HM-300 AC Power](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/159241718/c98f2b96-f160-4c79-bb1f-5b477cd3c0e4)
![P DC](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/159241718/137379db-5ed5-4c5c-8c45-36aa5538fd4c)
![P](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/159241718/526662ee-146e-44d4-91f2-62c32f4c8ff2)
Den zu beruhigen geht nur über Aus und wieder Einschalten des DPL per GUI. DC-seitiges Stromlosmachen hilft da nicht.
Ich versuche das nächste mal den Log mitzuschneiden. Es passiert sporadisch, da ist es schwer den Zeitpunkt zu treffen.
Ich bleibe dran.

**Comment by schlimmchen** [comment]
Und wie genau kommt der Power-Meter Wert in die OpenDTU? Vielleicht passiert ja doch was spannendes bei -2048 Watt?

**Reply by hoschiking** [reply]
Der kommt per HTTP(S) + JSON Abfrage, da ist mir nichts negatives aufgefallen.

**Comment by hoschiking** [comment]
Ich werde zusätzlich den DPL-Mode tracken und schauen, ob das Kommando zum Ausschalten wirklich ankommt. Das wäre noch eine mögliche Ursache, dass das der Mode 1 nicht gesetzt wird. Kurzer Test mit Modeumschaltung funktioniert soweit.
![DPL](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/159241718/04c1797f-91b9-4193-ae64-3012b744f1ad)

**Comment by schlimmchen** [comment]
@hoschiking Hast du schon 2024.03.07 installiert? Da wird ja sichergestellt, dass der Inverter den angestrebten Zustand wirklich erreicht, durch mehrfaches Absetzen entsprechender Kommandos. Gut möglich ist dein Problem damit behoben, weil die Nachricht in den Standby zu gehen wiederholt wird, wenn der Wechselrichter sie nicht gehört hat.

Es gibt zuletzt aber auch Berichte, dass Inverter neu gestartet werden müssen, damit sie Kommandos wieder verarbeiten. Dafür gibt es #748.

**Reply by hoschiking** [reply]
Klar ist das letzte Release installiert, allerdings habe ich seit dem diesen Fall nicht mehr beobachtet, auch vor dem neuesten Release. Ich habe aber auch letzten Samstag die Sendeleistung der DTU erhöht, die war auf kleinster Stufe und der neu hinzugekommene HM-600 auf dem Carport Dach zeitweise nicht erreichbar war. Zwei Mal ging der DPL auf Rot und ich konnte es nur durch Neustadt des ESP beheben. Nach der Erhöhung der Sendeleistung trat der Fehler bisher nicht auf. Dann habe ich noch gestern ve.direct deaktiviert, ich lade nur noch AC gestützt. In Summe gab es so viel Änderungen, dass es schwierig wird, auf die Ursache zu schließen, wenn der Fehler nicht mehr auftreten sollte. Was wirklich gut funktioniert, ist den SoC im DPL zu ignorieren. Die Änderung war richtig und wichtig.
